pandas/core/dtypes/inference.py

```diff
@@ -450,3 +450,38 @@ def is_sequence(obj):
         return not isinstance(obj, (str, bytes))
     except (TypeError, AttributeError):
         return False
+
+
+def is_dataclass_instance(item):
+    """
+    Checks if the object is a data-class instance
+
+    Parameters
+    ----------
+    item : object
+
+    Returns
+    --------
+    is_dataclass_instance : bool
+        True if the item is an instance of a data-class,
+        will return false if you pass the data class itself
+
+    Examples
+    --------
+    >>> @dataclass
+    >>> class Point:
+    ...     x: int
+    ...     y: int
+
+    >>> is_dataclass_instance(Point)
+    False
+    >>> is_dataclass_instance(Point(0,2))
+    True
+
+    """
+    try:
+        from dataclasses import is_dataclass
```

Comment:
I think we can actually put this at the top with a try/except around it

Reply:
That would require a global `HAS_DATCLASSES` variable, which I don't find as clean as this.